basics_of_sst-core/tutorial/Example00.md

```diff
@@ -0,0 +1,568 @@
+    logger_.verbose(CALL_INFO, INFO,  0x00, "Constructing new Example Instance.\n");
+
+    // Configure the component clock.
+    //
+    logger_.verbose(CALL_INFO, DEBUG, 0x00, "Clock rate is: %s\n", clock.c_str());
+    registerClock(clock,
+        new SST::Clock::Handler<ExampleComponent>(this, &ExampleComponent::clockTick));
+    logger_.verbose(CALL_INFO, INFO,  0x00, "Successfully initialized clock.\n");
+
+    // Register this component with the simulation.
+    //
+    registerAsPrimaryComponent();
+    logger_.verbose(CALL_INFO, DEBUG, 0x00, "Component registered as primary component.\n");
+    primaryComponentDoNotEndSim();
+    logger_.verbose(CALL_INFO, DEBUG, 0x00, "Simulation notified it should not end.\n");
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Leaving constructor for component id %lu\n", componentId_);
+}
+
+
+// Called after all components have been constructed and initialization
+// has completed, but before simulation time has begin.
+//
+// This is where you should do any other initialization that needs done
+// but could be accomplished in the constructure.
+//
+void ExampleComponent::setup(void)
+{
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Entering setup for component id %lu\n", componentId_);
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Leaving setup for component id %lu\n", componentId_);
+}
+
+
+// Called after the simulation is complete but before the objects are
+// destroyed.  This is a good place to print out statistics.
+//
+void ExampleComponent::finish(void)
+{
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Entering finish for component id %lu\n", componentId_);
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Leaving finish for component id %lu\n", componentId_);
+}
+
+
+// Clock event handler.
+//
+bool ExampleComponent::clockTick(SST::Cycle_t cycle)
+{
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Entering clock for component id %lu\n", componentId_);
+
+    // Increment the clock tick counter and end when you get to
+    // the specified value.
+    //
+    clockTickCount_ += 1;
+    bool done = (clockTickCount_ == clockTicks_);
+
+    logger_.verbose(CALL_INFO, INFO, 0x00, "Clock tick count = %lu out of %lu\n", clockTickCount_, clockTicks_);
+    if (done)
+    {
+        logger_.verbose(CALL_INFO, INFO, 0x00, "Ending sim.\n");
+        primaryComponentOKToEndSim();
+    }
+   
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Leaving clock for component id %lu\n", componentId_);
+    return done;
+}
+```
+
+### Understanding the Constructor
+
+For this component, all initialization takes place in the constructor.
+
+First, the component parameters are read from the configuration file.
+
+```
+   // Read in the parameters from the python config file.  See SST_ELI_DOCUMENT_PARAMS
+    // for an explanation of what each parameter represents.
+    //
+    std::string clock = 
+        params.find<std::string>("clock", "1GHz");
+    clockTicks_ = static_cast<uint64_t>(
```

Comment:
static_cast isn't needed here, you can just do params.find<uint64_t>("clockTicks", 10);